rcl_action/src/rcl_action/action_client.c

```diff
@@ -0,0 +1,812 @@
+  }
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Action result response taken");
+  return RCL_RET_OK;
+}
+
+rcl_ret_t
+rcl_action_send_cancel_request(
+  const rcl_action_client_t * action_client,
+  const void * ros_cancel_request)
+{
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Sending action cancel request");
+  if (!rcl_action_client_is_valid(action_client)) {
+    return RCL_RET_ACTION_CLIENT_INVALID;  // error already set
+  }
+  RCL_CHECK_ARGUMENT_FOR_NULL(ros_cancel_request, RCL_RET_INVALID_ARGUMENT);
+  int64_t ignored_sequence_number;
+  rcl_ret_t ret = rcl_send_request(
+    &action_client->impl->cancel_client,
+    ros_cancel_request, &ignored_sequence_number);
+  if (RCL_RET_OK != ret) {
+    return RCL_RET_ERROR;
+  }
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Action cancel request sent");
+  return RCL_RET_OK;
+}
+
+rcl_ret_t
+rcl_action_take_cancel_response(
+  const rcl_action_client_t * action_client,
+  void * ros_cancel_response)
+{
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Taking action cancel response");
+  if (!rcl_action_client_is_valid(action_client)) {
+    return RCL_RET_ACTION_CLIENT_INVALID;  // error already set
+  }
+  RCL_CHECK_ARGUMENT_FOR_NULL(ros_cancel_response, RCL_RET_INVALID_ARGUMENT);
+  rmw_request_id_t ignored_response_header;
+  rcl_ret_t ret = rcl_take_response(
+    &action_client->impl->cancel_client,
+    &ignored_response_header, ros_cancel_response);
+  if (RCL_RET_OK != ret) {
+    if (RCL_RET_CLIENT_TAKE_FAILED == ret) {
+      return RCL_RET_ACTION_CLIENT_TAKE_FAILED;
+    }
+    if (RCL_RET_BAD_ALLOC == ret) {
+      return RCL_RET_BAD_ALLOC;
+    }
+    return RCL_RET_ERROR;
+  }
+  RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Action cancel response taken");
+  return RCL_RET_OK;
+}
+
+const char *
+rcl_action_client_get_action_name(const rcl_action_client_t * action_client)
+{
+  if (!rcl_action_client_is_valid(action_client)) {
+    return NULL;
+  }
+  return action_client->impl->action_name;
+}
+
+const rcl_action_client_options_t *
+rcl_action_client_get_options(const rcl_action_client_t * action_client)
+{
+  if (!rcl_action_client_is_valid(action_client)) {
+    return NULL;
+  }
+  return &action_client->impl->options;
+}
+
+bool
+rcl_action_client_is_valid(const rcl_action_client_t * action_client)
+{
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    action_client, "action client pointer is invalid", return false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    action_client->impl, "action client implementation is invalid", return false);
+  return true;
+}
+
+rcl_ret_t
+rcl_action_wait_set_add_action_client(
+  rcl_wait_set_t * wait_set,
+  const rcl_action_client_t * action_client)
+{
+  rcl_ret_t ret;
+  RCL_CHECK_ARGUMENT_FOR_NULL(wait_set, RCL_RET_WAIT_SET_INVALID);
+  if (!rcl_action_client_is_valid(action_client)) {
+    return RCL_RET_ACTION_CLIENT_INVALID;  // error already set
+  }
+  // Wait on action goal service response messages.
+  ret = rcl_wait_set_add_client(
```

Comment:
nit: fit on single line. Same for other lines below.

Reply:
Fixed in 0c313821659b90f03650dfcefcbff453acba82f6.